### ISSUE TITLE
Prevent `C-RET` from moving forward to the next line 

### ISSUE THREAD

**Issue by amirteymuri** [jorgenschaefer/elpy#953]
This is not an issue but rather a question, i hope it is okey if i put it here!

Is it possible to prevent `C-RET` from moving to the next line? Mostly when i evaluate a line with `C-RET` i want to stay on the very same line and try it with some other arguments or what so ever, so every time having to hit the back button to go back to the line in question is a bit cumbersome (maybe i am a bit too lazy!).

**Comment by gopar** [comment]
`C-RET` is bind to `elpy-shell-send-current-statement` and the last line on that function is `python-nav-forward-statement`(which is the bit of code that moves forward). From what I can tell it looks like you can't disable it but you can simply copy the old function remove the last line and bind the func to `C-RET`.    

```lisp
(defun my-custom-elpy-shell-send-current-statement ()
  "Send current statement to Python shell."
  (interactive)
  (let ((beg (python-nav-beginning-of-statement))
        (end (python-nav-end-of-statement)))
    (elpy-shell-get-or-create-process)
    (python-shell-send-string (buffer-substring beg end)))
  (elpy-shell-display-buffer))

(define-key elpy-mode-map (kbd "C-RET") 'my-custom-elpy-shell-send-current-statement)
```

Or something like that. Hope this helps \o/

**Comment by amirteymuri** [comment]
I dont know any emacs lisp, but i suppose your code should go in the `init.el` file, right? I did so and no changes happen with C-ret

**Comment by gopar** [comment]
@amirteymuri  Ah, sorry! I thought you knew some emacs lisp aka elisp :)

Yeah, that bit of code should go in the `init.el`. After that, you can easiest thing you can do is restart emacs to see it take effect.

**Comment by amirteymuri** [comment]
As i said i have already added the code to init.el file and restarted emacs, but i cant see any changes. the  cursor still moves forward

**Comment by gopar** [comment]
Hmmm weird.

Would you happen to have an `init.elc` file in the same directory as `init.el`? If you do, you'll have delete the *.elc file. And try restarting again after that.

**Comment by amirteymuri** [comment]
No, my `init.el` file is in my `~/.emacs` directory, and there is no `.elc` file there.

**Comment by jguenther** [comment]
@amirteymuri: Try this:

```lisp
(with-eval-after-load 'elpy
  (define-key elpy-mode-map (kbd "C-RET") 'my-custom-elpy-shell-send-current-statement))
```

**Comment by amirteymuri** [comment]
@jguenther  Didn't help either.

**Comment by jguenther** [comment]
@amirteymuri what does `C-h k C-RET` show? It might be a binding in `python-mode-map` that's taking precedence.

How about:

``` lisp
(define-key elpy-mode-map (kbd "C-RET") 'my-custom-elpy-shell-send-current-statement)
(define-key python-mode-map (kbd "C-RET") nil)
```

**Comment by rgemulla** [comment]
This is possible with #1182.
